### PR TITLE
Github Action to tweet when labeled with tag "issue/tweet"

### LINKED ISSUE #6612: [CI] Setup GitHub Action to tweet when `issue/tweet` label is applied
#### Current Behavior
Many new and existing issues can benefit from being posted on Twitter for broader reach of potential contributors and for recognizing the efforts of existing contributors.

#### Desired Behavior
1. Upon application of the `issue/tweet` label, tweet link to issue from [mesheryio](https://twitter.com/mesheryio) account.

#### Implementation
1. Use https://github.com/marketplace/actions/send-tweet-action.
    1. `status:` should include the title of the [issue | pull request] - link-to-issue-or-pr #opensource 
    2. e.g. "[CI] Setup GitHub Action to tweet when `issue/tweet` label is applied - https://github.com/meshery/meshery/issues/6612 #opensource"
1. Request or create each necessary GitHub secret at the organization level.

#### Acceptance Tests
1. All `meshery` repos have `issue/tweet` label, which has color of #eee.
1. The [GitHub Process document](https://docs.google.com/document/d/1FzX-C_xy9hZ3Eu9dcCE0unhFV5LRUrb5YLn_MGYuG6Y/edit) is updated.
1. Label should work on both issues and pull requests.

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by gopivaibhav**
Signed-off-by: Gopi Vaibhav <gopivaibhav0@gmail.com>

**Notes for Reviewers**

We have to add these keys in the secrets in order to make this Github Action work:
- TWITTER_CONSUMER_API_KEY
- TWITTER_CONSUMER_API_SECRET
- TWITTER_ACCESS_TOKEN
- TWITTER_ACCESS_TOKEN_SECRET 

Video Proof of working after adding these secrets -
[Screen Recording by adding different tags to test](https://user-images.githubusercontent.com/78922463/209333896-76b25d3f-48d7-4f7f-bc8e-65a333ddf665.webm)

This PR fixes #6612 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by gopivaibhav** [comment]
Haii @leecalcote I have made the PR (including the demo video of sending a tweet from my account) with the required changes as discussed in the yesterday's meet but I am not sure of what I have to add in the tweet. So, I have added a demo message as of now and will change it according to the desired change.

**Comment by gopivaibhav** [comment]
@leecalcote I have made the commit which restricts the action to run only when members of meshery adds the `issue/tweet` label.
Video demo of the sample tweet with issue title and URL-
[Screencast from 12-25-2022 12:22:33 AM.webm](https://user-images.githubusercontent.com/78922463/209448236-ad93d5e9-2b9b-4ae2-97b3-ba051920863c.webm)

**Review by ayushthe1 — COMMENTED** [review]
hey in the issue it's mentioned that the workflow should run when label is applied either to issue or a pull request. So i think you should also add the support for pull request.

File: .github/workflows/label-tweet.yml
Comment (leecalcote) [inline .github/workflows/label-tweet.yml:16]: @gopivaibhav more than this, will you try using this - action https://github.com/tspascoal/get-user-teams-membership - to assess whether the user is a member of @meshery/maintainers?

**Comment by leecalcote** [comment]
Some good stuff in here, @gopivaibhav

File: .github/workflows/label-tweet.yml
Comment (gopivaibhav) [reply .github/workflows/label-tweet.yml:16]: Oh, okay. I will change it. Also, What am I supposed to tweet?
Is the content which I have written now looking good? Or do I have to change it to something else?

Comment (leecalcote) [reply .github/workflows/label-tweet.yml:16]: I missed your message here. Let's try this:

```
Help wanted: "<issue title>" - <issue link> #opensource
```

**Comment by Chadha93** [comment]
@gopivaibhav  Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by gopivaibhav** [comment]
> @gopivaibhav  Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

Sorry @Chadha93 I have been writing my exams for 5 days. I will be back again from 9th of January and will add this to the agenda of Meshery dev call then.

File: .github/workflows/label-tweet.yml
Comment (gopivaibhav) [reply .github/workflows/label-tweet.yml]: Added the required tweet body for the tweet

**Comment by gopivaibhav** [comment]
Hi @leecalcote @ayushthe1 I was busy with my exams. So, couldn't work more but I have updated the code to work for both issues and PRs. I also restricted only maintainers to tweet when labeled with `issue/tweet`. Let me know If I have to change anything.


Note - 
- We have to add a secret with the name PAT, which has the value of the GitHub token of this org with the scope- `read:org`
- I have added "maintainers" as the team name in lowercase but if the team name is "Maintainers", my code will result in a false for checking whether the actor is a maintainer or not. So, let me know the proper spelling of the team name with the correct case as I can't view the maintainers' team.

**Comment by leecalcote** [comment]
New org-wide secrets defined for consumer-key and consumer-secret.

**Comment by leecalcote** [comment]
> Hi @leecalcote @ayushthe1 I was busy with my exams. So, couldn't work more but I have updated the code to work for both issues and PRs. I also restricted only maintainers to tweet when labeled with `issue/tweet`. Let me know If I have to change anything.
> 
> Note -
> 
> * We have to add a secret with the name PAT, which has the value of the GitHub token of this org with the scope- `read:org`
> * I have added "maintainers" as the team name in lowercase but if the team name is "Maintainers", my code will result in a false for checking whether the actor is a maintainer or not. So, let me know the proper spelling of the team name with the correct case as I can't view the maintainers' team.

Great notes, @gopivaibhav. Thank you. I've shifted us from `PAT` to `GLOBAL_TOKEN` and have two of the four keys loaded thus far.

File: .github/workflows/label-tweet.yml
Comment (Nikhil-Ladha) [inline .github/workflows/label-tweet.yml:39]: nit: this

**Comment by ayushthe1** [comment]
hey @gopivaibhav ,now that your pr is almost complete ,could you also review mine at [here](https://github.com/layer5io/layer5/pull/3556) .

**Comment by Chadha93** [comment]
@gopivaibhav Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Review by Nikhil-Ladha — DISMISSED** [review]
lgtm

**Comment by Nikhil-Ladha** [comment]
Leaving the final call on @leecalcote to confirm the verbiage.

**Comment by gopivaibhav** [comment]
> hey @gopivaibhav ,now that your pr is almost complete ,could you also review mine at [here](https://github.com/layer5io/layer5/pull/3556) .

I have gone through your PR @ayushthe1 And it's more or less the same as the code I wrote. So, I guess there's nothing to review

File: .github/workflows/label-tweet.yml
Comment (leecalcote) [inline .github/workflows/label-tweet.yml:30]:
```suggestion
          status: 'Help Wanted: "${{ github.event.issue.title }}" - ${{ github.event.issue.html_url }} #opensource'
```

File: .github/workflows/label-tweet.yml
Comment (leecalcote) [inline .github/workflows/label-tweet.yml:39]:
```suggestion
          status: 'Help Wanted: "${{ github.event.pull_request.title }}" - ${{ github.event.pull_request.html_url }} #opensource'
```

**Comment by leecalcote** [comment]
> Leaving the final call on @leecalcote to confirm the verbiage.

Thank you, @Nikhil-Ladha. I've tweaked the verbiage on each tweet. ✅ 

We're without two of the necessary secrets just yet: API Token and Token Secret. Work to be done here.

**Comment by gopivaibhav** [comment]
Hi @leecalcote I am back to being active as I don't have any exams now. So, can you please let me know the status of this PR or if You are expecting anything else from me?

**Comment by Chadha93** [comment]
@gopivaibhav Let's discuss the update of this PR on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by gopivaibhav** [comment]
Hi @leecalcote Any update on this PR?

**Pull request merged**
